src/adapters/winston.js

```diff
@@ -13,25 +13,25 @@ const assert = require('assert');
   @function verbose - log message with a verbose log level
   @function log - log message with a provided log level
   @function profile - profiling function
-  @constructor consumes the instance of a LoggerChain @class
+  @constructor consumes the instance of a LoggerStream @class
 **/
 class Winston {
   /**
     @constructor
     Construct the instance of a winston adapter
     Create main logging functions
     Each function has the following arguments:
-    @param chain {object} - an instance of a chain
-    @param chainLinkIndex {number} - index of a chain link this adapter is for
+    @param stream {object} - an instance of a stream
+    @param subscriber {object} - instance of a subscriber
   **/
-  constructor(chain, chainLinkIndex = -1) {
-    assert(chain);
-    this.chain = chain;
-    this.chainLinkIndex = chainLinkIndex;
+  constructor(stream, subscriber) {
+    assert(stream);
+    this.stream = stream;
+    this.winston = subscriber.winston;
     const loggingLevels = ['error', 'warn', 'info', 'debug', 'silly', 'verbose'];
     for (const loggingLevel of loggingLevels) {
       this[loggingLevel] = (message, ...args) => {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define this function outside of a loop. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.dialonce.io/coding_rules#rule_key=javascript%3AFunctionDefinitionInsideLoop)